src/clocksi_interactive_coord.erl

```diff
@@ -564,6 +617,49 @@ execute_command(read_objects, Objects, Sender, State = #state{transaction=Transa
 
     {next_state, receive_read_objects_result, NewCoordState#state{from = Sender}};
 
+execute_command(validate_or_read, {Key, Type, Token}, Sender, State = #state{
+    transaction=Transaction,
```

Comment:
Would probably be more efficient to encode this as filter or fold over the `ReadResult` list.